actors/miner/tests/util.rs

```diff
@@ -2279,9 +2284,341 @@ fn fixed_hasher(offset: ChainEpoch) -> Box<dyn Fn(&[u8]) -> [u8; 32]> {
     Box::new(hash)
 }
 
+pub struct DealSummary {
+    pub sector_start: ChainEpoch,
+    pub sector_expiration: ChainEpoch,
+}
+
+pub struct StateSummary {
+    pub live_power: PowerPair,
+    pub active_power: PowerPair,
+    pub faulty_power: PowerPair,
+    pub deals: BTreeMap<DealID, DealSummary>,
+    pub window_post_proof_type: RegisteredPoStProof,
+    pub deadline_cron_active: bool,
+}
+
+impl Default for StateSummary {
+    fn default() -> Self {
+        StateSummary {
+            live_power: PowerPair::zero(),
+            active_power: PowerPair::zero(),
+            faulty_power: PowerPair::zero(),
+            window_post_proof_type: RegisteredPoStProof::Invalid(0),
+            deadline_cron_active: false,
+            deals: BTreeMap::new(),
+        }
+    }
+}
+
+fn check_miner_info(info: MinerInfo, acc: &MessageAccumulator) {
+    acc.require(
+        info.owner.protocol() == Protocol::ID,
+        &format!("owner address {} is not an ID address", info.owner),
+    );
+    acc.require(
+        info.worker.protocol() == Protocol::ID,
+        &format!("worker address {} is not an ID address", info.worker),
+    );
+    info.control_addresses.iter().for_each(|address| {
+        acc.require(
+            address.protocol() == Protocol::ID,
+            &format!("control address {} is not an ID address", address),
+        )
+    });
+
+    if let Some(pending_worker_key) = info.pending_worker_key {
+        acc.require(
+            pending_worker_key.new_worker.protocol() == Protocol::ID,
+            &format!(
+                "pending worker address {} is not an ID address",
+                pending_worker_key.new_worker
+            ),
+        );
+        acc.require(
+            pending_worker_key.new_worker != info.worker,
+            &format!(
+                "pending worker key {} is same as existing worker {}",
+                pending_worker_key.new_worker, info.worker
+            ),
+        );
+    }
+
+    if let Some(pending_owner_address) = info.pending_owner_address {
+        acc.require(
+            pending_owner_address.protocol() == Protocol::ID,
+            &format!("pending owner address {} is not an ID address", pending_owner_address),
+        );
+        acc.require(
+            pending_owner_address != info.owner,
+            &format!(
+                "pending owner address {} is same as existing owner {}",
+                pending_owner_address, info.owner
+            ),
+        );
+    }
+
+    if let RegisteredPoStProof::Invalid(id) = info.window_post_proof_type {
+        acc.add(&format!("invalid Window PoSt proof type {id}"));
+    };
+}
```

Comment:
Two checks missing: 1. that `info.window_post_proof_type.sector_size() == info.sector_size` 2. that `info.window_post_partitions_sector() == info.window_post_partition_sectors`

Reply:
thanks!